hr_shift/views/shift_planning_views.xml

```diff
@@ -0,0 +1,575 @@
+                        type="object"
+                        name="generate_shifts"
+                        class="btn btn-primary"
+                        string="Generate shifts"
+                        invisible="state != 'new' or not end_date"
+                    />
+                    <button
+                        type="object"
+                        name="copy_to_planning"
+                        class="btn btn-secondary"
+                        string="Copy to new planning"
+                        invisible="state == 'new' or not shift_ids"
+                    />
+                    <button
+                        type="object"
+                        name="regenerate_shifts"
+                        class="btn btn-secondary"
+                        string="Regenerate shifts"
+                        invisible="state != 'assignment' or not shift_ids"
+                        confirm="This will remove the current shifts assignations for this planning"
+                    />
+                </header>
+                <sheet>
+                    <div class="oe_button_box" name="button_box">
+                        <button
+                            class="oe_stat_button"
+                            name="action_view_shifts"
+                            type="object"
+                            invisible="state == 'new'"
+                            icon="fa-link"
+                        >
+                            <field
+                                name="shifts_count"
+                                string="Shifts"
+                                widget="statinfo"
+                            />
+                        </button>
+                        <button
+                            class="oe_stat_button"
+                            name="action_view_issued_shifts"
+                            type="object"
+                            invisible="state == 'new' or issued_shifts_count == 0"
+                            icon="fa-unlink"
+                        >
+                            <field
+                                name="issued_shifts_count"
+                                string="Issues"
+                                widget="statinfo"
+                            />
+                        </button>
+                    </div>
+                    <group>
+                        <group>
+                            <field name="year" readonly="state != 'new'" />
+                            <field name="week_number" readonly="state != 'new'" />
+                        </group>
+                        <group>
+                            <field name="start_date" readonly="state != 'new'" />
+                            <field name="end_date" force_save="1" />
+                        </group>
+                    </group>
+                </sheet>
+            </form>
+        </field>
+    </record>
+    <record id="shift_planning_kanban" model="ir.ui.view">
```

Comment:
In v15, in the kanban view, the “_undefined_” column was displayed by default. I think this should be kept to avoid confusion.

**v15**
<img width="1234" height="467" alt="ejemplo-v15" src="https://github.com/user-attachments/assets/793fbc6c-fec4-4d81-83f5-41e0672972a5" />

**v17**
<img width="958" height="301" alt="ejemplo-v17" src="https://github.com/user-attachments/assets/f2f943f6-d391-46d3-bc0a-2cbb6830d437" />